docs/how-to/package-with-hatch.md

```diff
@@ -0,0 +1,380 @@
+```
+
+Each entry in the `[project.scripts]` section (there can be more than one) maps a command on to an import and function name.
+In the second line above, before the `=` character, `calculator` is the name of the command.
+The string contains the name of the import (`textual_calculator.entry_points`) followed by a colon (`:`) and the name of a function (also called `calculator`).
+
+It is equivalent to doing the following from the Python REPL:
+
+```
+>>> import textual_calculator.entry_points
+>>> textual_calculator.entry_points.calculator()
+```
+
+To add the `calculator` command once you have edited `pyproject.toml`, run the following from the command line:
+
+```bash
+pip install -e .
+```
+
+Now you can launch the calculator from the command line as follows:
+
+```bash
+calculator
+```
+
+## Building 
+
+Building produces archive files that contain your code.
+When you install a package via pip or other tool, it will download one of these archives.
+
+To build your project with Hatch, change to the directory containing your `pyproject.toml` and run the `hatch build` subcommand:
+
+```
+cd textual-calculator
+hatch build
+```
+
+After a moment, you should find that Hatch has created a `dist` (distribution) folder, which contains the project archive files.
+You don't typically need to use these files directly, but feel free to have a look at the directory contents.
+
+!!! note "Packaging TCSS and other files"
+
+    Hatch will typically include all the files needed by your project, i.e. the `.py` files.
+    It will also include any Textual CSS (`.tcss`) files in the project directory.
+    Not all build tools will include files other than `.py`; if you are using another build tool, you may have to consult the documentation for how to add the Textual CSS files.
+
+
+## Publishing
+
+After your project has been successfully built you are ready to publish it to PyPi.
+
+If you don't have a PyPi account, you can [create one now](https://pypi.org/account/register/).
+Be sure to follow the instructions to validate your email and set up 2FA (Two Factor Authentication).
+
+Once you have an account, login to PyPi and go to the Account Settings tab.
+Scroll down and click the "Add API token" button.
+In the "Create API Token" form, create a token with name "Uploads" and select the "Entire project" scope, then click the "Create token" button.
+
+Copy this API token (long string of random looking characters) somewhere safe.
+This API token is how PyPi authenticates uploads are for your account, so you should never share your API token or upload it to the internet.
+
+Run the following command (replacing `<YOUR API TOKEN>` with the text generated in the previous step):
+
+```bash
+hatch publish -u __token__ -a <YOUR API TOKEN>
+```
+
+Hatch will upload the distribution files, and you should see a PyPi URL in the terminal.
+
+### Managing API Tokens
+
+Creating an API token for all projects is required for the first upload.
```

Comment:
I don't think it's clear that “all projects” here refers to the _permissions_ of the token.